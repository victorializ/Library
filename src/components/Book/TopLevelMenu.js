import React  from 'react';
import { Search, Button } from 'semantic-ui-react'

import { MenuButton } from './MenuButton';
import { constants } from '../../i18n';
import { book, sorting } from '../../types';
import { colors } from '../../assets/semantic-colors';

const TopLevelMenu = ({
  filter, 
  onFilterSet, 
  input, 
  onInput, 
  sort, 
  onSortSet, 
  onSortUnsert, 
  onFilterUnsert
}) =>
  <div className="books__menu">
    <div className="books__search">
      <Button.Group>
        <Button 
          onClick={onFilterUnsert}
          color={colors.secondary}>
          {constants.filter}
        </Button>
        <MenuButton 
          name={book.name} 
          selectedButton={filter} 
          clickHandler={onFilterSet} />
        <Button.Or />
        <MenuButton 
          name={book.bookYear} 
          selectedButton={filter} 
          clickHandler={onFilterSet} />
        <Button.Or />
        <MenuButton 
          name={book.bookAuthors} 
          selectedButton={filter} 
          clickHandler={onFilterSet} />
        <Button.Or />
        <MenuButton 
          name={book.bookGenres} 
          selectedButton={filter} 
          clickHandler={onFilterSet} />
      </Button.Group>
      <Search
        value = {input}
        showNoResults={false}
        onSearchChange ={(e, {value}) => onInput(value)}
      ></Search>
    </div>
    <div className="books__filter">
      <Button.Group>
        <Button 
          onClick={onSortUnsert}
          color={colors.secondary}>
          {constants.sort}
        </Button>
        <MenuButton 
          name={sorting.asc} 
          selectedButton={sort} 
          clickHandler={onSortSet} />
        <Button.Or />
        <MenuButton 
          name={sorting.desc} 
          selectedButton={sort} 
          clickHandler={onSortSet} />
      </Button.Group>
    </div>
  </div>

export { TopLevelMenu as BooksTopLevelMenu };