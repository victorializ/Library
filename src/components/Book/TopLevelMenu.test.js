import React from 'react';
import { shallow } from 'enzyme';

import { BooksTopLevelMenu } from './TopLevelMenu';

describe('TopLevelMenu component', () => {
    it('renders top level menu component correctly', () => {
        expect(shallow(<BooksTopLevelMenu />)).toMatchSnapshot();
    });
});