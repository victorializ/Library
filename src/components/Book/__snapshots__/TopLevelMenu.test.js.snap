// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TopLevelMenu component renders top level menu component correctly 1`] = `
<div
  className="books__menu"
>
  <div
    className="books__search"
  >
    <ButtonGroup>
      <Button
        as="button"
        className="books__reset-filter"
        color="black"
      >
        Filters
      </Button>
      <MenuButton
        name="name"
      />
      <ButtonOr />
      <MenuButton
        name="bookYear"
      />
      <ButtonOr />
      <MenuButton
        name="bookAuthors"
      />
      <ButtonOr />
      <MenuButton
        name="bookGenres"
      />
    </ButtonGroup>
    <Search
      icon="search"
      input="text"
      minCharacters={1}
      noResultsMessage="No results found."
      onSearchChange={[Function]}
      showNoResults={false}
    />
  </div>
  <div
    className="books__filter"
  >
    <ButtonGroup>
      <Button
        as="button"
        className="books__reset-filter"
        color="black"
      >
        Sort
      </Button>
      <MenuButton
        name="asc"
      />
      <ButtonOr />
      <MenuButton
        name="desc"
      />
    </ButtonGroup>
  </div>
</div>
`;
